### PR TITLE
Feature/refactoring to response handlers

### PR DESCRIPTION
**Pull request by guyv**
added response handler concept

added jsfiles/modules support

it's hacky, should be refactored

### CODE REVIEW

File: lib/components/proxy-request-handler.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 'use strict';
 
 var httpProxy = require('http-proxy');
+var _ = require('lodash');
 var contracts = require('../../api/contracts');
 
 var targetAddress = 'http://localhost:8080/';
 var proxy;
 
+var responseHandlers = [];
```
Comment (Vrolijkx) [inline lib/components/proxy-request-handler.js:10]: shoudn't this be part of the proxyRequestHandler?

Comment (guyv) [inline lib/components/proxy-request-handler.js:10]: all components are singletons, unless they are defined as a function
I didn't want to expose these as state.

File: lib/components/translations-response-handler.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+var _ = require('lodash');
+var url = require('url');
+var contracts = require('../../api/contracts');
+
+var projects = [];
```
Comment (Vrolijkx) [inline lib/components/translations-response-handler.js:7]: idem

File: lib/components/Project.js [hunk 3]
```diff
@@ -60,6 +62,24 @@ var Project = function () {
 			});
 			console.log('Updating translations for ' + projectDir);
 		},
+		updateUiFiles: function(category, uiFiles, extension) {
+			var files = _.flatMap(uiFiles, function(uiFile) {
```
Comment (Vrolijkx) [inline lib/components/Project.js]: maybe use shorthand `_.flatMap(uiFiles, 'value')`

Comment (guyv) [inline lib/components/Project.js]: done

**Pull request merged**